realm/realm-library/src/syncIntegrationTest/java/io/realm/SSLConfigurationTests.java

```diff
@@ -133,28 +136,34 @@ public void trustedRootCA_syncShouldFailWithoutTrustedCA() throws InterruptedExc
         SyncUser user = SyncUser.login(SyncCredentials.usernamePassword(username, password, true), Constants.AUTH_URL);
 
         // 1. Copy a valid Realm to the server
-        final SyncConfiguration configOld = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM)
+        //noinspection unchecked
+        final SyncConfiguration syncConfig = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM)
                 .schema(StringOnly.class)
                 .build();
-        Realm realm = Realm.getInstance(configOld);
+        Realm realm = Realm.getInstance(syncConfig);
 
         realm.beginTransaction();
         realm.createObject(StringOnly.class).setChars("Foo");
         realm.commitTransaction();
 
         // make sure the changes gets to the server
-        SystemClock.sleep(TimeUnit.SECONDS.toMillis(2));  // FIXME: Replace with Sync Progress Notifications once available.
+        SyncManager.getSession(syncConfig).uploadAllLocalChanges();
         realm.close();
         user.logout();
 
         // 2. Local state should now be completely reset. Open the Realm again with a new configuration which should
         // download the uploaded changes.
         user = SyncUser.login(SyncCredentials.usernamePassword(username, password), Constants.AUTH_URL);
-        SyncConfiguration config = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM_SECURE)
+        //noinspection unchecked
+        SyncConfiguration syncConfigSSL = configurationFactory.createSyncConfigurationBuilder(user, Constants.USER_REALM_SECURE)
                 .name("useSsl")
                 .schema(StringOnly.class)
+                .trustedRootCA("untrusted_ca.pem")
+//                .waitForInitialRemoteData()
                 .build();
-        realm = Realm.getInstance(config);
+        // waitForInitialRemoteData will throw an Internal error (125): Operation Canceled
```

Comment:
because the SSL handshake fails. This will trigger the following stacktrace causing the internal error 

```
E/REALM_SYNC( 2147): Connection[2]: SSL handshake failed: certificate verify failed
I/REALM_SYNC( 2147): Connection[2]: Connection closed due to error
D/REALM_JNI( 2147): error_handler lambda invoked
E/REALM_JAVA( 2147): Session Error[realms://127.0.0.1:9443/8f5b98a75b5d7ee90be1afe105711242/tests]: SSL_SERVER_CERT_REJECTED(117)
E/REALM_JAVA( 2147): SSL server certificate rejected
D/REALM_SYNC( 2147): Connection[2]: Reconnecting in 872 milliseconds
I/REALM_SYNC( 2147): Closing Realm file: /data/data/io.realm.test/cache/junit1611252515/trustedRootCA_syncShouldFailWithoutTrustedCA(io.realm.SSLConfigurationTests)/8f5b98a75b5d7ee90be1afe105711242/8f5b98a75b5d7ee90be1afe105711242/useSsl
I/TestRunner( 2147): failed: trustedRootCA_syncShouldFailWithoutTrustedCA(io.realm.SSLConfigurationTests)
I/TestRunner( 2147): ----- begin exception -----
I/TestRunner( 2147): UNKNOWN(-1)
I/TestRunner( 2147): Internal error (125): Operation Canceled
I/TestRunner( 2147): 	at io.realm.SyncSession$WaitForSessionWrapper.throwExceptionIfNeeded(SyncSession.java:728)
I/TestRunner( 2147): 	at io.realm.SyncSession.waitForChanges(SyncSession.java:427)
I/TestRunner( 2147): 	at io.realm.SyncSession.downloadAllServerChanges(SyncSession.java:353)
I/TestRunner( 2147): 	at io.realm.internal.SyncObjectServerFacade.downloadRemoteChanges(SyncObjectServerFacade.java:166)
I/TestRunner( 2147): 	at io.realm.RealmCache.doCreateRealmOrGetFromCache(RealmCache.java:303)
I/TestRunner( 2147): 	at io.realm.RealmCache.createRealmOrGetFromCache(RealmCache.java:282)
I/TestRunner( 2147): 	at io.realm.Realm.getInstance(Realm.java:353)
I/TestRunner( 2147): 	at io.realm.SSLConfigurationTests.trustedRootCA_syncShouldFailWithoutTrustedCA(SSLConfigurationTests.java:166)
```